### PR TITLE
Add usePrevious React hook to Platform

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
I propose we have a place for React Hooks in Platform

## Testing done
- [x] Unit tests

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by sanlouise — APPROVED** [review]
Looks great to me! Thanks for adding this as a reusable hook 👍

File: src/platform/utilities/react-hooks.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      valueRef.current = value;
+    },
+    // only run this useEffect hook if `value` changes
+    [value],
+  );
+  // this returns _before_ the `useEffect` call completes, so this will return
+  // whatever was stored in `ref.current` _before_ this function's `useEffect`
+  // callback runs
+  return valueRef.current;
+};
```
Comment (ncksllvn) [inline src/platform/utilities/react-hooks.js:27]: Does it make me old by thinking the React lifecycle hooks are easier to understand? 🤷

**Review by ncksllvn — APPROVED** [review]

File: src/platform/utilities/react-hooks.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * documentation:
+ * https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state
+ *
+ * as well as on usehooks.com:
+ * https://usehooks.com/usePrevious/
+ *
+ * This post does a deeper dive for those who are new to hooks:
+ * https://blog.logrocket.com/how-to-get-previous-props-state-with-react-hooks/
+ */
+export const usePrevious = value => {
```
Comment (theodur) [inline src/platform/utilities/react-hooks.js:14]: Can you add jsdoc comments for this hook?

**Review by theodur — APPROVED** [review]
This looks good to me!

**Pull request merged**